playbook.yml

```diff
@@ -16,15 +16,43 @@
     shell:
        sed -i 's/REGION/eu-west-2/' /etc/yum.repos.d/redhat-rhui*.repo
 
+  - name: perform dnf group install
+    shell:
+       dnf group install 'Development Tools' -y 
+
   - name: DNF Install Dependencies
     dnf: name={{ item }}
     with_items:
       - zip
       - unzip
       - git
       - python39
+      - wget
+      - perl-core
+      - zlib-devel
     become: true
 
+  - name: Attempting to install perl-IPC-Cmd through shell
+    shell:
+      dnf install perl-IPC-Cmd -y
+
+  - name: get openssl and build from source
+    shell: |
+      cd /tmp && wget https://www.openssl.org/source/openssl-3.0.9.tar.gz && tar -xzvf openssl-*.tar.gz && rm -rf openssl-*.tar.gz 
+      cd /tmp/openssl-* && ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl shared zlib
```

Comment:
Executing `cd` to a directory with `*` is bad practice. 
That wildcard could expand to multiple values and `cd` would be confused.
It would be better to use the actual directory name, if it is possible to know it in advance.